models/engine/db_storage.py

```diff
@@ -74,3 +74,25 @@ def reload(self):
     def close(self):
         """call remove() method on the private session attribute"""
         self.__session.remove()
+
+    def get(self, cls, id):
+        """Returns the object based on the class name and its ID """
+        cls_dict = {}
+        cls_dict = self.all(cls)
+
+        if cls is not None:
+            cls_dict = self.all(cls)
+            for obj in cls_dict.values():
+                if obj.id == id:
+                    return obj
+        return None
+
+    def count(self, cls=None):
+        """
+           Returns the number of objects in storage matching the given
+           class name.
+        """
+        cls_dict = {}
```

Comment:
Initializing `cls_dict` seems unnecessary since right after that you'll be receiving the type of whatever the function `self.all(cls)` returns. Yet, I think it is a good practice to be explicit on what you expect to receive by initializing like so. It makes the code more readable.